### PR TITLE
SCSS variables and mixins cleanup

### LINKED ISSUE #11458: Stylesheets variables and mixins clean-up
Follow-up to #11365 - there are a number of quick win improvements with how we write Sass:

- Remove the unused `$grid-content-indent` and `$grid-max-width`.
- Move `$grid-columns` and `$grid-gutter-width` to the one mixin file they’re used in. I don’t see ourselves reusing those outside that file ever.
- Replace $font-sans with direct usage of `theme('fontFamily.sans')`, possibly re-testing the different places it’s used as I suspect lots of them are completely unnecessary.
- Replace $font-mono with direct usage of `theme('fontFamily.mono')`
- Move `$sidebar-toggle-size` and `$sidebar-toggle-spacing` to the one file they’re used in.
- Remove all unused mixins in `_mixins.grid.scss`
- Remove the unused `$object-title-height`

### Working on this

Anyone can contribute to this. View our [contributing guidelines](https://docs.wagtail.org/en/latest/contributing/index.html), add a comment to the issue once you’re ready to start. This is pretty mechanical work but in CSS / Sass only, so requires an awareness of the syntax and how to test different types of changes in Wagtail.

### PR DESCRIPTION
**Pull request by Jaivignesh-afk**
Fixes #11458 

Not a duplicate of #11468
Check comments in the issue #11458

1.Removed unused $grid-content-indent,$grid-max-width and $object-title-height.
2.Replaced $font-sans with direct usage of theme('fontFamily.sans') and $font-mono with direct usage of theme('fontFamily.mono').
3.Moved $grid-columns,$grid-gutter-width,$sidebar-toggle-size and $sidebar-toggle-spacing to the file that they are used in.
4.Removed all unused mixins.

_Please check the following:_

-   [X] Do the tests still pass?[^1]
-   [X] Does the code comply with the style guide?
    -   [X] Run `make lint` from the Wagtail root.
-   [ ] For Python changes: Have you added tests to cover the new/fixed behaviour?
-   [ ] For front-end changes: Did you test on all of Wagtail’s supported environments?[^2]
    -   [ ] **Please list the exact browser and operating system versions you tested**:
    -   [ ] **Please list which assistive technologies [^3] you tested**:
-   [ ] For new features: Has the documentation been updated accordingly?

**Please describe additional details for testing this change**.

[^1]: [Development Testing](https://docs.wagtail.org/en/latest/contributing/developing.html#testing)
[^2]: [Browser and device support](https://docs.wagtail.org/en/latest/contributing/developing.html#browser-and-device-support)
[^3]: [Accessibility Target](https://docs.wagtail.org/en/latest/contributing/developing.html#accessibility-targets)

### CODE REVIEW

**Comment by lb-** [comment]
@Jaivignesh-afk this looks like almost an exact copy of the PR you referenced. I'll close this PR for now. 

If you have something to add to that PR may I suggest you add a comment to it. 

Or if there is something majorly different about this PR that should mean it replaces the other, please let us know.


https://github.com/wagtail/wagtail/pull/11468

**Comment by Jaivignesh-afk** [comment]
Apologies for the oversight, it seems there was a mistake in referencing the issue number in this PR.I've removed some unused mixins which were not removed in this PR#11468.

**Comment by lb-** [comment]
Ok. Maybe it's best to add a comment to the other PR with what to remove and where.

Alternatively, adjust this PR to be just that part that got missed and update the description and we can re-open. 

Thanks for the clarification.

**Comment by Jaivignesh-afk** [comment]
Actually I was the first one to work on this issue.@nandini584 didn't check if someone was working on this issue before sending a PR. So that's the reason I'm kind of duplicating the PR.You can check comments in this issue#11458. But still if you want, we can continue with PR#11468. I will comment there regarding the changes.

**Comment by lb-** [comment]
Ok. Thanks for the context @Jaivignesh-afk - we're all trying to work together here and I'm sure it was just a bit of confusion. 

In the future, just help us out with info on the description of your PR. 

I'll try to review this PR and the other related ones together. I'll re-open this for now.

**Comment by Jaivignesh-afk** [comment]
Thanks for reopening the issue. I have updated the description of this PR providing info and will do so in the future.

File: client/scss/settings/_variables.scss [hunk 1]
```diff
@@ -1,19 +1,11 @@
 @use 'sass:color';
 // paths
-
 // We can't use absolute paths here, because those are dependent on Django's
 // STATIC_URL setting. Instead, relative paths from the final location of the
 // compiled CSS files are used.
 // */
-
```
Comment (lb-) [inline client/scss/settings/_variables.scss:8]: Probably best to leave this line break in, but we can remove the `// */` above.

File: client/scss/settings/_variables.scss [hunk 2]
```diff
@@ -1,19 +1,11 @@
...
 $images-root: '../images/';
 
-// grid settings
-$grid-columns: 12;
-$grid-gutter-width: 3%;
-$grid-max-width: 1200px;
-$grid-content-indent: 0.7;
-
 $mobile-nice-padding: 20px;
 $desktop-nice-padding: 80px;
```
Comment (lb-) [inline client/scss/settings/_variables.scss:10]: Might be nice to update these two variables to their linked them sizing while we are here.


```
$mobile-nice-padding: theme('spacing.5');
$desktop-nice-padding: theme('spacing.20');
```

**Review by lb- — APPROVED** [review]
Thanks @Jaivignesh-afk  I have reviewed these changes alongside the changes in #11468 from @nandini584 

Together these are looking good and I will push up a few tweaks and merge in shortly.

1. Updated variables `$mobile-nice-padding` & `$desktop-nice-padding`
2. I found a few more unused variables with this epic library https://github.com/XhmikosR/find-unused-sass-variables & have removed them.

FYI @thibaudcolas

**Comment by lb-** [comment]
@Jaivignesh-afk  I have added you to the contributors list as `Jai Vignesh J`, please let me know if you wanted a different name listed.

**Comment by Jaivignesh-afk** [comment]
Thank you for reviewing my PR @lb- . Jai Vignesh J is fine.

**Pull request merged**